Improve page rendering by deferring disqus loading

KaitoHH opened the pull request:
<!-- ATTENTION!
1. Please, write pull request readme in English. Not all contributors / collaborators know Chinese and Google translate can't always translate issues accurately. Thanks!

2. Always remember that NexT includes 4 schemes. And if on one of them works fine after the changes, on another scheme this changes can be broken. Muse and Mist have similar structure, but Pisces is very difference from them. Gemini is a mirror of Pisces with some styles and layouts remakes. So, please, make the tests at least on two schemes (Muse or Mist and Pisces or Gemini).
-->

## PR Checklist
**Please check if your PR fulfills the following requirements:**
<!-- Change [ ] to [X] to select -->

- [X] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [X] Tests for the changes was maked (for bug fixes / features).
   - [X] Muse | Mist have been tested.
   - [X] Pisces | Gemini have been tested.
- [ ] [Docs](https://github.com/theme-next/theme-next.org/tree/source/source/docs) in [NexT website](https://theme-next.org/docs/) have been added / updated (for features).
<!-- For adding Docs edit needed file here: https://github.com/theme-next/theme-next.org/tree/source/source/docs and create PR with this changes here: https://github.com/theme-next/theme-next.org/pulls -->

## PR Type
**What kind of change does this PR introduce?**

- [ ] Bugfix.
- [ ] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [X] Other... Please describe: **Improvement**

## What is the current behavior?
The whole web page is blocked and shows nothing until loading `embed.js` for Disqus is completed. Things get worse when some users (especially those Chinese users) have trouble connecting Disqus since they will see a blank page for a long time until the requested timeout.

![image](https://user-images.githubusercontent.com/13927774/58751403-7c995b80-84d0-11e9-86f7-95c5fcca1810.png)

In addition, for those who enable `pace.js`, they will see the progress bar circling for a long time until timeout.

## What is the new behavior?
The main cause of this problem is that loading `*.js` for Disqus blocks the rendering of the whole web page, so I simply defer the js loading until the whole page loading is completed.

https://github.com/theme-next/hexo-theme-next/blob/54a04c6209b60b554a47f19523553b0dda3c5336/layout/_third-party/comments/disqus.swig#L10

Here I use window.onload callback function instead of directly inserting script tag to prevent blocking the whole web page.

## Does this PR introduce a breaking change?
- [ ] Yes.
- [X] No.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/theme-next/hexo-theme-next?pullRequest=891) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/theme-next/hexo-theme-next?pullRequest=891) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/theme-next/hexo-theme-next?pullRequest=891) it.</sub>

stevenjoezhang commented:
How about disqusJS? https://github.com/theme-next/hexo-theme-next/pull/705

KaitoHH commented:
Fixed. Notice that DisqusJS also have the same problem as #395. So I fixed it by adding essential url and identifier attribute.

https://github.com/theme-next/hexo-theme-next/blob/98c30836868d6fee0072be30fa2757d058dd0b53/layout/_third-party/comments/disqusjs.swig#L18-L19

1v9 commented:
@allcontributorsadd please add @KaitoHH for code
